app/src/utils/RedirectionUtils.js

```diff
@@ -25,9 +26,12 @@ export const redirectToRules = (navigate, hardRedirect) => {
 };
 
 /* FEATURE - RULES - Create New Rule */
-export const redirectToCreateNewRule = (navigate, ruleType) => {
+export const redirectToCreateNewRule = (navigate, ruleType, source) => {
   if (ruleType) {
-    navigate(`${PATHS.RULE_EDITOR.CREATE_RULE.ABSOLUTE}/${ruleType}`);
+    navigate({
+      pathname: `${PATHS.RULE_EDITOR.CREATE_RULE.ABSOLUTE}/${ruleType}`,
+      search: `?${createSearchParams({ source })}`,
+    });
```

Comment:
We should not use `query params` for inapp navigation. Lets use navigation `state` instead.